C++/C.sublime-syntax

```diff
@@ -429,125 +482,312 @@ contexts:
+      push:
+        - meta_scope: debug.variable-identifier-list.array-or-assignment
+        - include: declaration-brackets
+        - match: \=
+          scope: keyword.operator.assignment.c
+          set:
+            - meta_scope: debug.variable-identifier-list.assignment
+            - match: '(?=,|;)'
+              pop: true
+            - include: expressions
+            - include: or_pop
+        - include: or_pop
+    - match: ','
+      scope: punctuation.separator.c debug.variable-identifier-list
+    - match: ';'
+      scope: punctuation.terminator.c
       pop: true
+    - include: or_pop
 
-  data-structures-struct-definition:
-    - meta_scope: meta.struct.c
-    - include: data-structures-definition-common-begin
-    - match: '{{identifier}}(?=\s*;)'
-      scope: entity.name.struct.forward-decl.c
+  member-variable-identifier-list:
+    - meta_scope: debug.member-variable-identifier-list
+    - include: comments
+    - include: modifiers
+    - include: modifiers-parens
+    - include: modifiers-macros
+    - match: \*
+      scope: storage.modifier.c
+    - match: \(
+      scope: punctuation.section.group.begin.c debug.member-variable-identifier-list.function-pointer
+      comment: Function pointer declaration
+      push:
+        - meta_scope: meta.parens
+        - match: \)
+          scope: punctuation.section.parens.end
+          set:
+            - include: function-definition-params-group
+            - include: or_pop
+        - match: \*
+          scope: storage.modifier.c
+        - match: '{{identifier}}'
+          scope: entity.name.member.indexed.c debug.member-variable-identifier-list.function-pointer
+    - match: '{{common_identifier}}'
+      scope: entity.name.member.c debug.member-variable-identifier-list
+      comment: There are a lot of common variable names 4 characters or under.
+      push:
+        - meta_scope: debug.member-variable-identifier-list.array
+        - include: declaration-brackets
+        - include: or_pop
     - match: '{{identifier}}'
-      scope: entity.name.struct.c
-    - match: '(?=;)'
+      scope: entity.name.member.indexed.c debug.member-variable-identifier-list
+      push:
+        - meta_scope: debug.member-variable-identifier-list.array
+        - include: declaration-brackets
+        - include: or_pop
+    - match: ','
+      scope: punctuation.separator.c debug.member-variable-identifier-list
+    - match: ';'
+      scope: punctuation.terminator.c
       pop: true
+    - include: or_pop
+
+  struct-keyword:
+    - match: '\bstruct\b'
+      scope: storage.type.c
+      push:
+        - meta_scope: meta.struct.c debug.struct-keyword
+        - include: data-structures-definition-common-begin
+        - match: '{{identifier}}(?=\s*{|\s*\n)'
+          scope: entity.name.struct.c
+          comment: Struct definition
+          set: struct-body
+        - match: '{{identifier}}'
+          scope: support.type.c
```

Comment:
`support.type` is usually used for built-in or framework identifiers. Is this intended to match macro calls?

Reply:
What is the scope I should use for referencing user defined types? In the java syntax I used `support.class`

Reply:
Looking at the tests might make it more clear.

```
struct foo { <-- This is an entity
 ...
};
```

vs

```
void func(struct bar *ptr); <-- This is a `support.type.c`
```

I used the term declaration because I'm declaring the type of a variable. Open to suggestions.

It sounds like we have new scope keywords though. I can get those updated.